src/core/webview/ClineProvider.ts

```diff
@@ -2130,262 +2065,113 @@ export class ClineProvider implements vscode.WebviewViewProvider {
-			this.getGlobalState("customSupportPrompts") as Promise<CustomSupportPrompts | undefined>,
-			this.getGlobalState("enhancementApiConfigId") as Promise<string | undefined>,
-			this.getGlobalState("autoApprovalEnabled") as Promise<boolean | undefined>,
-			this.customModesManager.getCustomModes(),
-			this.getGlobalState("experiments") as Promise<Record<ExperimentId, boolean> | undefined>,
-			this.getSecret("unboundApiKey") as Promise<string | undefined>,
-			this.getGlobalState("unboundModelId") as Promise<string | undefined>,
-			this.getGlobalState("unboundModelInfo") as Promise<ModelInfo | undefined>,
-			this.getSecret("requestyApiKey") as Promise<string | undefined>,
-			this.getGlobalState("requestyModelId") as Promise<string | undefined>,
-			this.getGlobalState("requestyModelInfo") as Promise<ModelInfo | undefined>,
-			this.getGlobalState("modelTemperature") as Promise<number | undefined>,
-			this.getGlobalState("modelMaxTokens") as Promise<number | undefined>,
-			this.getGlobalState("anthropicThinking") as Promise<number | undefined>,
-			this.getGlobalState("maxOpenTabsContext") as Promise<number | undefined>,
+		// Create an object to store all fetched values
+		const stateValues: Record<GlobalStateKey | SecretKey, any> = {} as Record<GlobalStateKey | SecretKey, any>
+		const secretValues: Record<SecretKey, any> = {} as Record<SecretKey, any>
+
+		// Create promise arrays for global state and secrets
+		const statePromises = GLOBAL_STATE_KEYS.map((key) => this.getGlobalState(key))
+		const secretPromises = SECRET_KEYS.map((key) => this.getSecret(key))
+
+		// Add promise for custom modes which is handled separately
+		const customModesPromise = this.customModesManager.getCustomModes()
+
+		// Wait for all promises to resolve
+		const [stateResults, secretResults, customModes] = await Promise.all([
+			Promise.all(statePromises),
+			Promise.all(secretPromises),
+			customModesPromise,
 		])
 
+		// Populate stateValues and secretValues
+		GLOBAL_STATE_KEYS.forEach((key, index) => {
+			stateValues[key] = stateResults[index]
+		})
+
+		SECRET_KEYS.forEach((key, index) => {
+			secretValues[key] = secretResults[index]
+		})
+
+		// Determine apiProvider with the same logic as before
 		let apiProvider: ApiProvider
-		if (storedApiProvider) {
-			apiProvider = storedApiProvider
+		if (stateValues.apiProvider) {
+			apiProvider = stateValues.apiProvider
 		} else {
 			// Either new user or legacy user that doesn't have the apiProvider stored in state
 			// (If they're using OpenRouter or Bedrock, then apiProvider state will exist)
-			if (apiKey) {
+			if (secretValues.apiKey) {
 				apiProvider = "anthropic"
 			} else {
 				// New users should default to openrouter
 				apiProvider = "openrouter"
 			}
 		}
 
+		// Build the apiConfiguration object combining state values and secrets
+		const apiConfiguration: ApiConfiguration = {
+			apiProvider,
+			apiModelId: stateValues.apiModelId,
```

Comment:
Is there some way to use the globalstatekeys here so we don't need to repeat this big list?

Reply:
Let me think that way

Reply:
@mrubens for this, i try to do it but it make this pr become so big,  i think should we do it in another pr ?

Reply:
@mrubens  i update this new commit for this comment, can you check it again